### PR TITLE
Sample disposal20180904

### PR DESCRIPTION
**Pull request by EGCGDanForth**
prepare_discard_plate updated to use re search to find full container name format to introduce the possibility of searching for plates ending in P999.

remove_from_freezer  updated as previously updated the submitted sample freezer location for derived samples. Now checks if sample is a submitted sample or a derived sample and updates only the corresponding freezer location UDFs.

### CODE REVIEW

File: scripts/prepare_discard_plate.py [hunk 1]
```diff
@@ -1,16 +1,23 @@
 #!/usr/bin/env python
+import re
 import sys
+from logging import FileHandler
```
Comment (mwhamgenomics) [inline scripts/prepare_discard_plate.py]: Unused import

File: scripts/prepare_discard_plate.py [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
+
 from EPPs.common import StepEPP, get_workflow_stage
+from egcg_core.app_logging import logging_default as log_cfg
```
Comment (mwhamgenomics) [inline scripts/prepare_discard_plate.py]: Unused import

File: scripts/prepare_discard_plate.py [hunk 3]
```diff
@@ -98,14 +107,16 @@ def _run(self):
...
                     # This container will have to wait
                     route_allowed = False
                     self.info(
                         "Container: %s won't route because artifact %s in step_associated_artifacts (%s) or has been discarded before (%s)",
-                        container.name, artifact.name, artifact in step_associated_artifacts, has_workflow_stage(artifact, sample_discard_wf_stage_name)
+                        container.name, artifact.name, artifact in step_associated_artifacts,
+                        has_workflow_stage(artifact, sample_discard_wf_stage_name)
                     )
             if route_allowed:
                 artifacts_to_route.extend(list(container.placements.values()))
```
Comment (mwhamgenomics) [inline scripts/prepare_discard_plate.py]: Print statement below is redundant and can be removed

Comment (EGCGDanForth) [reply scripts/prepare_discard_plate.py]: removed

**Comment by EGCGDanForth** [comment]
Batch call of samples add to remove_from_freezer.py per request from tcezard

File: scripts/prepare_discard_plate.py [hunk 4]
```diff
@@ -51,6 +51,7 @@ def has_workflow_stage(artifact, workflow_step_name):
 
 
 class FindPlateToRoute(StepEPP):
+    _use_load_config = False
```
Comment (mwhamgenomics) [inline scripts/prepare_discard_plate.py]: This needs the discussed changes in `EPPs.common.StepEPP` in order to be meaningful. This PR could then close #67

Comment (EGCGDanForth) [reply scripts/prepare_discard_plate.py]: I guess either Tim or yourself will implement this change?

**Comment by EGCGDanForth** [comment]
LIMS Change Log CR757

**Pull request merged**